osm2streets/src/road_parts.rs

```diff
@@ -39,41 +38,22 @@ pub enum RoadRanks {
 /// A usage designation for an area, such as a lane.
 #[derive(Clone, Debug, PartialEq)]
 pub enum Designation {
-    /// Think "carriageway" carriages, anything from trucks to mopeds to drawn carts.
-    Travel(Carriage),
-    /// Areas of the road that are explicitly not for (normal) driving.
-    /// Often painted, sometimes w/barriers.
+    /// A part of the road designated for travel.
+    Travel {
+        carriage: Carriage,
+        direction: TrafficDirections,
+    },
+    /// A part of the road designated for parking / "standing".
+    Parking { carriage: Carriage },
+    /// A part of the road that is explicitly not (normally) for carriages.
+    /// E.g. a painted buffer, median or verge.
```

Comment:
This subsumes what amenity used to be?

Reply:
Yeah, I don't remember my justification for being more specific than `NoTravel` at this level, and I'm trying not to get too carried away with datatypes that I'm not using yet. In any case, physical features should probably be details *of* `NoTravel`, instead of a distinct variant.